packages/core/src/particle/ParticleRenderer.ts

```diff
@@ -343,11 +350,16 @@ export class ParticleRenderer extends GeometryRenderer {
     this._isStart = false;
     this._isInit = false;
     this._maxCount = value;
+    this._updateDirtyFlag = DirtyFlagType.Everything;
     this.geometry = this._createGeometry();
 
     this._updateBuffer();
 
     this._isInit = true;
+
+    if (this._isAutoplay) {
```

Comment:
isAutoplay should call play on onEnable event,should not call on set maxCount. 
The name can more detail,for exmaple "playOnEnabled".